packages/reassure-compare/src/utils/validate.ts

```diff
@@ -0,0 +1,35 @@
+import { z } from 'zod';
+import { hasDuplicateValues } from './array';
+
+const metadataSchema = z.object({
```

Comment:
We would want to no longer have separate type definitions as we would need to manual sync these with zod. A better approach would be to use zod infering:

```suggestion
const performanceMetadataSchema = z.object({ ... });
type PerformanceMetadata = z.infer<typeof performanceMetadataSchema>;
```

Reply:
We will export these types and use them in `types.ts` file right @mdjastrzebski ?

Reply:
We cannot replace `PerformanceMetadataSchema` directly as its present state does not have a top level `metadata` tag.

Reply:
Let's add both `PerformanceMetadataScheme` (=== PerformanceMetadata type)
and `PerformanceHeaderScheme` (PerformanceHeader type)